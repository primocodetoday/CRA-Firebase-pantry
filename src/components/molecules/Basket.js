import React from 'react';
import styled from 'styled-components';
import Title from '../atoms/Title';

const StyledWrapper = styled.div`
  background-color: white;
  display: flex;
  flex-direction: column;
  flex-grow: 1;
  color: ${({ theme }) => theme.secondary};
  margin: 15px 0px 15px 0;
  border-radius: 15px;
  flex-basis: 370px;
  min-height: 250px;
  padding: 30px;
`;

const Line = styled.div`
  background-color: ${({ theme }) => theme.primary};
  height: 1px;
  width: 100%;
`;

const Basket = () => {
  return (
    <StyledWrapper>
      <Title>ShopList</Title>
      <Line />
      <p>List</p>
    </StyledWrapper>
  );
};

export default Basket;
