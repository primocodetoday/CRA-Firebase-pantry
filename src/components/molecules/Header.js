import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Title, SubTitle } from '../atoms';

const StyledWrapper = styled.header`
  width: 100%;
  margin: 25px 0;

  @media (max-width: ${({ theme }) => theme.mediaBreaks.mobile}px) {
    text-align: center;
  }
`;

const StyledTitle = styled(Title)`
  color: white;
`;
const StyledSubTitle = styled(SubTitle)`
  color: white;
  margin-top: 17px;
`;

const Header = ({ titleText, subTitleText }) => {
  return (
    <StyledWrapper>
      <StyledTitle>{titleText}</StyledTitle>
      <StyledSubTitle>{subTitleText}</StyledSubTitle>
    </StyledWrapper>
  );
};

Header.propTypes = {
  titleText: PropTypes.string.isRequired,
  subTitleText: PropTypes.string.isRequired,
};

export default Header;
