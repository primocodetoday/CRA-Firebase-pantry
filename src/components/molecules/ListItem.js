import React, { useState } from 'react';
import styled from 'styled-components';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import Paragraph from '../atoms/Paragraph/Paragraph';

const StyledWrapper = styled.li`
  width: 100%;
  display: grid;
  grid-template-columns: 1fr 35px 25px;
  margin: 5px 0;
  cursor: pointer;
  color: ${(theme) => theme.primary};

  @media (max-width: ${({ theme }) => theme.mediaBreaks.mobile}px) {
    font-size: 2.3rem;
  }
`;

const StyledStock = styled(Paragraph)`
  text-align: right;
  margin-right: 3px;
`;

const ListItem = ({ name, stock, id, unit }) => {
  const [isRedirect, setRedirect] = useState({ redirect: false });

  const handleItemClick = () => setRedirect({ redirect: true });

  if (isRedirect.redirect) {
    return <Redirect to={`/pantry/${id}`} />;
  }
  return (
    <StyledWrapper onClick={handleItemClick}>
      <Paragraph>{name}</Paragraph>
      <StyledStock>{stock}</StyledStock>
      <Paragraph>{unit}</Paragraph>
    </StyledWrapper>
  );
};

ListItem.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  stock: PropTypes.number.isRequired,
  unit: PropTypes.string.isRequired,
};

export default ListItem;
