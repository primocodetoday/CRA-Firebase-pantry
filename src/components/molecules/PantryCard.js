import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Title from '../atoms/Title';

const Wrapper = styled.div`
  position: relative;
  background-color: white;
  color: ${({ theme }) => theme.primary};
  border-radius: ${({ theme }) => theme.borderRadius};
  flex-basis: 400px;
  min-height: 250px;
  padding: 37px 30px;
  margin-bottom: 25px;
  margin-right: 52px;
  flex-grow: 1;
`;

const TitleCard = styled(Title)`
  text-align: right;
  font-size: ${({ theme }) => theme.fontSize.big};
  margin-bottom: 25px;
`;

const Line = styled.div`
  background-color: #123c69;
  height: 1px;
  width: 100%;
`;

const CardIcon = styled.div`
  position: absolute;
  top: 22px;
  left: 32px;
  width: 55px;
  height: 70px;

  background-image: url(${({ icon }) => icon});
  background-repeat: no-repeat;
  background-position: 50% 50%;
  background-size: contain;
`;

const PantryCard = ({ category, icon }) => {
  return (
    <Wrapper>
      <CardIcon icon={icon} />
      <TitleCard>{category}</TitleCard>
      <Line />
      <p>List</p>
    </Wrapper>
  );
};

PantryCard.propTypes = {
  category: PropTypes.string.isRequired,
  icon: PropTypes.string.isRequired,
};

export default PantryCard;
