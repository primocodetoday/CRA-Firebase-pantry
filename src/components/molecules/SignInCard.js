import styled from 'styled-components';

const SignInCard = styled.div`
  padding: 20px;
  background-color: white;
  width: 450px;
  border-radius: ${({ theme }) => theme.borderRadius};
  position: absolute;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
  display: grid;
  align-items: center;
  justify-items: center;
  grid-template-rows: 100px 40px 1fr 5px 1fr 5px 80px;

  @media (max-width: ${({ theme }) => theme.mediaBreaks.mobile}px) {
    width: 90vw;
  }
`;

export default SignInCard;
