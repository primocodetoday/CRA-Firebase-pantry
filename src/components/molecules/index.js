export { Basket } from './Basket/Basket';
export { CardList } from './CardList/CardList';
export { Header } from './Header/Header';
export { ListItem } from './ListItem/ListItem';
