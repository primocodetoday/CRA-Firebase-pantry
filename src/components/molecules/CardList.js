import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import ListItem from '../atoms/ListItem';

const StyledWrapper = styled.ul`
  display: flex;
  flex-direction: column;
  width: 100%;
  padding-left: 0 !important;
  margin: 6px 0;
`;

const CardList = ({ content }) => {
  const list = content.map(({ id, name, stock, unit }) => (
    <ListItem key={id} id={id} name={name} stock={stock} unit={unit} />
  ));

  return <StyledWrapper>{list}</StyledWrapper>;
};

CardList.propTypes = {
  content: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default CardList;
