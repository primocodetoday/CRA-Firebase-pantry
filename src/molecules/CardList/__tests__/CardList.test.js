import React from 'react';
import { render } from '@testing-library/react';
import { CardList } from 'molecules';

describe('<CardList/> renders properly>', () => {
  const content = [{ id: '1', name: 'Bread', stock: 15, unit: 'kg' }];

  it('with  passed props', () => {
    const { getByText, container } = render(<CardList content={content} />);

    expect(getByText('Bread')).toBeTruthy();
    expect(getByText('15')).toBeTruthy();
    expect(getByText('kg')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
