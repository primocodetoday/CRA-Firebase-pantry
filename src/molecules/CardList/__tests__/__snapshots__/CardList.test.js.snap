// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList/> renders properly> with  passed props 1`] = `
<ul
  class="StyledCardList__Wrapper-zp7mvs-0 gBOpsV"
>
  <li
    class="StyledListItem__Wrapper-m6611x-0 ffofAU"
  >
    <p
      class="Paragraph-vj49lv-0 eenHSs"
    >
      Bread
    </p>
    <p
      class="Paragraph-vj49lv-0 StyledListItem__Stock-m6611x-1 eenHSs eMTcUE"
    >
      15
    </p>
    <p
      class="Paragraph-vj49lv-0 eenHSs"
    >
      kg
    </p>
  </li>
</ul>
`;
